### ISSUE TITLE
Could you say more about what you are trying to run cpu-only? Pretraining, fine-tuning, embedding, etc.

### ISSUE THREAD

**Issue by Kevin-chen-sheng** [songlab-cal/tape#9]
Could you say more about what you are trying to run cpu-only? Pretraining, fine-tuning, embedding, etc.

_Originally posted by @nickbhat in https://github.com/songlab-cal/tape/issues/2#issuecomment-566204058_

**Comment by Kevin-chen-sheng** [comment]
> seq1

- `MSGHSKWATTKHKKAAIDAKRGKLFARLIKNIEVAARLGGGDPSGNPTLYDAIQKAKKSSVPNDNITRAVKRGSGEGADAVNYETIMYEAYGPAGVAILIECLTDNRNRAVSDVRVAVTRNGGTMADGGSVQRLFQRKGVVSVSKTYEVE`

My sequence format, shown above (fasta), has two categories.Now, I want to use your code as a classifier, which of your five tasks can be my reference?
Thanks in advance.

**Comment by rmrao** [comment]
I'm not entirely sure what you mean. I think you're saying that you have two types of proteins and are trying to classify / cluster them.

If you have labels for some of your data, you could set up a fine-tuning task and train a classifier directly. Otherwise, you could start from just the language modeling embeddings, and seeing if your proteins cluster.

We currently do not provide pretrained versions of all the downstream tasks in pytorch. We may do so in the future. You can of course download the data and train the model yourself. It is possible that e.g. using a model pretrained on contact prediction will provide better clustering b/c it has been trained pull out structure.

**Comment by Kevin-chen-sheng** [comment]
To be precise, I just want to add a new task with the ultimate goal of using your model for protein sequence classification.

**Comment by rmrao** [comment]
Ah ok - you can look at the example [here](https://github.com/songlab-cal/tape/blob/master/examples/adding_task.py) for the general procedure of adding a task.

A classification task is most similar to the remote homology task, which you can use as a reference. The only differences will be 1) Load in your own data / labels 2) register the task with only 2 classes instead of 1195 in remote homology.

**Comment by Kevin-chen-sheng** [comment]
I now have two fasta files as training data, one is a positive class and the other is a negative class.Another fasta file is used as a test file.But now I don't know how to construct the form of the data that would make it the right data for this work.
In addition, where can I see the data format for remote homology?

**Comment by rmrao** [comment]
You can download the remote homology data from [here](http://s3.amazonaws.com/proteindata/data_pytorch/remote_homology.tar.gz). This is the LMDB data - which is not a text file format, so you won't be able to directly look at it. We also provide it as JSON data [here](http://s3.amazonaws.com/proteindata/data_raw/remote_homology.tar.gz). There are links to all datasets in the README and you can also run the `download_data.sh` script to download everything.

In order to use this data you need to create a Pytorch Dataset (see [here](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html)). We provide some utilities for loading fasta, json, and lmdb data, but in general don't make any requirements as to how your data is stored / how you load it.

So major steps would be 

1. Create a pytorch dataset that returns examples (positive **and** negative) from your files
2. Make sure you can load both a training dataset (when the argument `split='train'` is passed in) and a validation dataset (when the argument `split='val'` is passed in).
3. Write a collate function that takes in a batch of your data and pads it and returns a dictionary with three keys: 'input_ids', 'input_mask', and 'targets'. I think you can just copy the `collate_fn` of the remote homology dataset w/o modification.

**Comment by Kevin-chen-sheng** [comment]
![image](https://user-images.githubusercontent.com/52196486/71413359-d0e27380-268c-11ea-8bab-c4118ea30249.png)
data/PDB/PDB_train.fasta
data/PDB/PDB_test.fasta
registry.register_task_model('PDB', 'transformer', ProteinBertForSequenceToSequenceClassification)
I followed your instructions and used adding_task.py directly, but got an error.Since I don't have pdb_valid.fasta in my file, why is it 'unrecognized split :valid'?

**Comment by Kevin-chen-sheng** [comment]
![image](https://user-images.githubusercontent.com/52196486/71470127-2a21e280-2806-11ea-99b2-2b435d7082e9.png)
I added a task named my_pdb according to your instructions, but this error occurred when running tape-train.

**Comment by rmrao** [comment]
Were you able to run the example adding task?

**Comment by Kevin-chen-sheng** [comment]
When I executed this command, python adding_task.py transformer secondary_structure_8, the following error (oom) was reported;When I set --batch_size 2, it works fine.

![image](https://user-images.githubusercontent.com/52196486/71569455-32319900-2b0a-11ea-81db-fd0ef30991e3.png)

**Comment by rmrao** [comment]
Awesome, sounds like everything is working then. Will close this issue.
